reconcile/ldap_users.py

```diff
@@ -1,78 +1,64 @@
 import logging
 from collections import defaultdict
+from dataclasses import dataclass
+from itertools import starmap
 
 from reconcile import (
     mr_client_gateway,
-    queries,
 )
-from reconcile.utils import gql
+from reconcile.gql_definitions.common.users_paths import UserV1
+from reconcile.typed_queries.ldap_settings import get_ldap_settings
+from reconcile.typed_queries.users_paths import get_users_paths
 from reconcile.utils.defer import defer
 from reconcile.utils.ldap_client import LdapClient
 from reconcile.utils.mr import (
     CreateDeleteUserAppInterface,
     CreateDeleteUserInfra,
 )
-from reconcile.utils.mr.user_maintenance import PathTypes
+from reconcile.utils.mr.user_maintenance import PathSpec, PathTypes
 
 QONTRACT_INTEGRATION = "ldap-users"
 
 
-def init_users() -> list[dict[str, list]]:
-    app_int_users = queries.get_users(refs=True)
+@dataclass
+class UserPaths:
+    username: str
+    paths: list[PathSpec]
 
-    users = defaultdict(list)
-    for user in app_int_users:
-        u = user["org_username"]
-        item = {"type": PathTypes.USER, "path": "data" + user["path"]}
-        users[u].append(item)
-        for r in user.get("requests"):
-            item = {"type": PathTypes.REQUEST, "path": "data" + r["path"]}
-            users[u].append(item)
-        for q in user.get("queries"):
-            item = {"type": PathTypes.QUERY, "path": "data" + q["path"]}
-            users[u].append(item)
-        for g in user.get("gabi_instances"):
-            item = {"type": PathTypes.GABI, "path": "data" + g["path"]}
-            users[u].append(item)
-        for a in user.get("aws_accounts", []):
-            item = {"type": PathTypes.AWS_ACCOUNTS, "path": "data" + a["path"]}
-            users[u].append(item)
-        for s in user.get("schedules"):
-            item = {"type": PathTypes.SCHEDULE, "path": "data" + s["path"]}
-            users[u].append(item)
-
-    return [{"username": username, "paths": paths} for username, paths in users.items()]
-
-
-LDAP_SETTINGS_QUERY = """
-{
-  settings: app_interface_settings_v1 {
-    ldap {
-      serverUrl
-      baseDn
-    }
-  }
-}
-"""
 
-
-def get_ldap_settings() -> dict:
-    """Returns LDAP settings"""
-    gqlapi = gql.get_api()
-    settings = gqlapi.query(LDAP_SETTINGS_QUERY)["settings"]
-    if settings:
-        # assuming a single settings file for now
-        return settings[0]
-    raise ValueError("no app-interface-settings settings found")
+def transform_users_paths(users_paths: list[UserV1]) -> list[UserPaths]:
+    users = defaultdict(list)
```

Comment:
Please don't use `starmap` and this temporary dict. Something like 

```
for user in users_paths:
    up = UsersPaths(username=user.org.username)
    up.paths.append(PathSpec(PathTypes.USER, user.path))
    ...
```
would be more readable